drivers/setup/blue/blue.c

```diff
@@ -1209,15 +1456,53 @@ ScrIoControl(
             PUCHAR Src, Dest;
             UINT32 SrcDelta, DestDelta, i;
 
+            /* Validate output buffer */
+            if (stk->Parameters.DeviceIoControl.OutputBufferLength < sizeof(CONSOLE_DRAW))
+            {
+                Status = STATUS_INVALID_PARAMETER;
+                break;
+            }
+            ASSERT(Irp->MdlAddress);
+            ConsoleDraw = (PCONSOLE_DRAW)MmGetSystemAddressForMdl(Irp->MdlAddress);
+            if (ConsoleDraw == NULL)
+            {
+                Status = STATUS_INSUFFICIENT_RESOURCES;
+                break;
+            }
+
+            /* Check whether we have the size for the header plus the data area */
+            if (stk->Parameters.DeviceIoControl.OutputBufferLength <
+                    sizeof(CONSOLE_DRAW) + ConsoleDraw->SizeX * ConsoleDraw->SizeY * 2)
+            {
+                Status = STATUS_INVALID_BUFFER_SIZE;
+                break;
+            }
+
+            // TODO: For the moment if the ConsoleDraw rectangle has borders
+            // out of the screen-buffer we just bail out. Would it be better
+            // to actually clip the rectangle within its borders instead?
+            if ( ConsoleDraw->X < 0 || ConsoleDraw->X >= DeviceExtension->Columns ||
+                 ConsoleDraw->Y < 0 || ConsoleDraw->Y >= DeviceExtension->Rows )
+            {
+                Status = STATUS_SUCCESS;
+                break;
+            }
+            if ( ConsoleDraw->X + ConsoleDraw->SizeX >= DeviceExtension->Columns ||
+                 ConsoleDraw->Y + ConsoleDraw->SizeY >= DeviceExtension->Rows )
```

Comment:
Almost! You still need a check for overflow.
E.g. `X = 10`, `SizeX = 0xFFFFFFFF`, the addition would come out as `9` which would be less than `Columns`.

Reply:
Should be fixed now.